File: apps/mybookkeeper/frontend/src/shared/types/welcome-manual/welcome-manual-send-response.ts
import type { WelcomeManualSendStatus } from "./welcome-manual-send-status";

/**
 * Mirrors backend `WelcomeManualSendResponse`. Returned at HTTP 200 even on a
 * failed/skipped send — ``status`` communicates the outcome so the email
 * dialog can render the correct result step.
 */
export interface WelcomeManualSendResponse {
  id: string;
  manual_id: string;
  recipient_email: string;
  recipient_name: string | null;
  status: WelcomeManualSendStatus;
  error_reason: string | null;
  created_at: string;
}

File: apps/mybookkeeper/backend/app/schemas/welcome_manuals/welcome_manual_send_response.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WelcomeManualSendResponse(BaseModel):
    """Send-log record returned to the HOST after a send attempt.

    Returning ``recipient_email``/``recipient_name`` is NOT cross-tenant
    leakage — the host owns the manual and just typed these values. ``status``
    communicates the outcome (sent / failed / skipped) so the frontend can
    render a clear success or couldn't-send message; ``error_reason`` is a
    short diagnostic for failed/skipped attempts.
    """

    id: uuid.UUID
    manual_id: uuid.UUID
    recipient_email: str
    recipient_name: str | None = None
    status: str
    error_reason: str | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
